worker/raft/rafttransport/worker.go

```diff
@@ -88,20 +73,29 @@ func NewWorker(config Config) (*Worker, error) {
 		return nil, errors.Trace(err)
 	}
 
+	certPool, err := api.CreateCertPool(config.APIInfo.CACert)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	apiPorts := config.APIInfo.Ports()
+	if n := len(apiPorts); n != 1 {
+		return nil, errors.Errorf("api.Info has %d unique ports, expected 1", n)
+	}
+
 	w := &Worker{
-		config:         config,
-		connections:    make(chan net.Conn),
-		dialRequests:   make(chan dialRequest),
-		serversUpdated: make(chan struct{}),
+		config:       config,
+		connections:  make(chan net.Conn),
+		dialRequests: make(chan dialRequest),
+		tlsConfig:    api.NewTLSConfig(certPool),
+		apiPort:      apiPorts[0],
 	}
 
 	const logPrefix = "[transport] "
 	logWriter := &raftutil.LoggoWriter{logger, loggo.DEBUG}
 	logLogger := log.New(logWriter, logPrefix, 0)
 	transport := raft.NewNetworkTransportWithConfig(&raft.NetworkTransportConfig{
-		Logger:                logLogger,
-		MaxPool:               7, // 7 is the maximum juju cluster size
-		ServerAddressProvider: serverAddressProvider{},
+		Logger:  logLogger,
+		MaxPool: 7, // 7 is the maximum juju cluster size
```

Comment:
Is this "7" a const somewhere?

Reply:
replicaset.MaxPeers. Didn't really want to bring in that dependency, but I've done it.